### PR TITLE
[TASK -2] Setup Route and Create Login Screens

### PR DESCRIPTION
**Pull request by rafayet-monon**
## What happened
 **[Feature Card](https://github.com/rafayet-monon/surveyor/projects/1#card-43745422)**

- Setup Routes
- Setup Auth Context
- Implement Login screens
 
## Insight

- [x] Setup Routes
- [x] Setup Authentication Context
- [x] Setup Bootstrap
- [x] Implement background layout
- [x] Implement logo component
- [x] Implement form components
- [x] Add Unit Tests
- [x] Add Cypress plugin

## Proof Of Work
- **Expected Design from [Figma](https://www.figma.com/file/i1AI0IzCWuJApX5CK0DmiW/Challenge---Web-App?node-id=6%3A168)**  

Login - Filled

- **Implemented Design**

Screenshot 2020-08-24 at 8 31 22 AM

- **Responsiveness**

Screenshot 2020-08-24 at 8 35 42 AM

- **Test Cases**

This CI should pass

### CODE REVIEW

**Comment by rafayetn** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  
<b>🃏 FAIL</b> in <a href="https://github.com/rafayet-monon/surveyor/blob/feature/implement-authentication-template/src/App.test.js">App.test.js</a>


<li>
<a href='https://github.com/rafayet-monon/surveyor/blob/feature/implement-authentication-template/src/App.test.js#L8'>renders app name SURVEYOR</a>
<br/>
<body> <div> <div> <div class="background-layout" > <div class="background-overlay" > <div class="container" > <div class="login-wrapper" > <div class="logo-label-wrapper" > NIMBLE <span class="text-label" > Sign in to Nimble </span> </div> <div class="form-wrapper" > <form> <div> <div class="text-input-wrapper" > <label class="input-label" for="email" > Email </label> <input class="form-control input-control" id="email" type="email" /> </div> </div> <div> <div class="text-input-wrapper" > <label class="input-label" for="password" > Password </label> <input class="form-control input-control" id="password" type="password" /> <a class="forgot-pass" href="/" > Forgot? </a> </div> </div> <div> <div class="button-wrapper" > <button class="btn button-custom" type="submit" > Sign in </button> </div> </div> </form> </div> </div> </div> </div> </div> </div> </div></body>

<details>
<summary>Full message</summary>
<pre><code>
Error: Unable to find an element with the text: /SURVEYOR/i. This could be because the text is broken up by multiple elements. In this case, you can provide a function for your text matcher to make your matcher more flexible.

<body>
  <div>
    <div>
      <div
        class="background-layout"
      >
        <div
          class="background-overlay"
        >
          <div
            class="container"
          >
            <div
              class="login-wrapper"
            >
              <div
                class="logo-label-wrapper"
              >
                NIMBLE
                <span
                  class="text-label"
                >
                  Sign in to Nimble
                </span>
              </div>
              <div
                class="form-wrapper"
              >
                <form>
                  <div>
                    <div
                      class="text-input-wrapper"
                    >
                      <label
                        class="input-label"
                        for="email"
                      >
                        Email
                      </label>
                      <input
                        class="form-control input-control"
                        id="email"
                        type="email"
                      />
                    </div>
                  </div>
                  <div>
                    <div
                      class="text-input-wrapper"
                    >
                      <label
                        class="input-label"
                        for="password"
                      >
                        Password
                      </label>
                      <input
                        class="form-control input-control"
                        id="password"
                        type="password"
                      />
                      <a
                        class="forgot-pass"
                        href="/"
                      >
                        Forgot?
                      </a>
                    </div>
                  </div>
                  <div>
                    <div
                      class="button-wrapper"
                    >
                      <button
                        class="btn button-custom"
                        type="submit"
                      >
                        Sign in
                      </button>
                    </div>
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
    at Object.getElementError (/home/runner/work/surveyor/surveyor/node_modules/@testing-library/dom/dist/config.js:34:12)
    at /home/runner/work/surveyor/surveyor/node_modules/@testing-library/dom/dist/query-helpers.js:71:38
    at getByText (/home/runner/work/surveyor/surveyor/node_modules/@testing-library/dom/dist/query-helpers.js:54:17)
    at Object.<anonymous> (/home/runner/work/surveyor/surveyor/src/App.test.js:8:21)
    at Object.asyncJestTest (/home/runner/work/surveyor/surveyor/node_modules/jest-jasmine2/build/jasmineAsyncInstall.js:102:37)
    at /home/runner/work/surveyor/surveyor/node_modules/jest-jasmine2/build/queueRunner.js:43:12
    at new Promise (<anonymous>)
    at mapper (/home/runner/work/surveyor/surveyor/node_modules/jest-jasmine2/build/queueRunner.js:26:19)
    at /home/runner/work/surveyor/surveyor/node_modules/jest-jasmine2/build/queueRunner.js:73:41
</code></pre>
</details>
</li>
<br/>


  </td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against eb41093f304734647ba338d838e53804cc503635
</p>

**Comment by Lahphim** [comment]
@rafayet-monon I think later on we can break down this card to be smaller 🤔
For example, we can break `bootstrap` and `cypress` into some `chore` tasks 👍

But for now, let's keep like this.

File: src/App.scss [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+// Globals
+@import 'stylesheets/base/body';
+@import 'stylesheets/base/basic';
+@import 'stylesheets/base/global';
+// Components
+@import 'stylesheets/components/background-layout';
+@import 'stylesheets/components/logo-with-label';
+@import 'stylesheets/components/text-input';
+@import 'stylesheets/components/form-button';
+// Containers
```
Comment (Lahphim) [inline src/App.scss:14]: @rafayet-monon I think this `container` is using in react structure.
As I see in these files it seems like we're using for the specific screen of layout 🤔
It'd be better if we keep the folder structure for styling to look like this below.

Screen Shot 2563-08-25 at 10 50 42

File: src/App.scss [hunk 2]
```diff
@@ -0,0 +1,17 @@
+// Variables
+@import 'stylesheets/variables/color-variables';
```
Comment (Lahphim) [inline src/App.scss:2]: Screen Shot 2563-08-25 at 10 55 33

@rafayet-monon I think we should move this one out of the folder and re-name it to `_variables.scss` 🤔

File: src/App.scss [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+// Mixins
+@import 'stylesheets/mixins/font-mixins';
+// Globals
+@import 'stylesheets/base/body';
+@import 'stylesheets/base/basic';
+@import 'stylesheets/base/global';
+// Components
+@import 'stylesheets/components/background-layout';
+@import 'stylesheets/components/logo-with-label';
```
Comment (Lahphim) [inline src/App.scss:11]: @rafayet-monon I think this component should be just `logo` and for the `label` will use it as a modifier.

http://smacss.com/
Screen Shot 2563-08-25 at 10 59 16

**Comment by rafayet-monon** [comment]
Closing PR due to large changes.

**Pull request closed without merging**